### ISSUE TITLE
Staging: Invalid response, header: HTTP/1.1 405 Method Not Allowed

### ISSUE THREAD

**Issue by gilbitron** [yourivw/LEClient#81]
Using `https://acme-staging-v02.api.letsencrypt.org` I'm getting the following response:

```
Invalid response, header: HTTP/1.1 405 Method Not Allowed
Server: nginx
Date: Fri, 13 Dec 2019 11:12:32 GMT
Content-Type: application/problem+json
Content-Length: 103
Connection: keep-alive
Cache-Control: public, max-age=0, no-cache
Link: <https://acme-staging-v02.api.letsencrypt.org/directory>;rel="index"
```

It looks like, as of the 4th Dec 2019, [unauthenticated `GET` requests have been deprectaed](https://community.letsencrypt.org/t/acme-v2-scheduled-deprecation-of-unauthenticated-resource-gets/74380) for the staging API. The same will happen for the production API on November 1st, 2020.

**Comment by surfskidude** [comment]
I get the same , but my client works with non-staging acme-v02.api.letsencrypt.org

For staging, I get:
https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/28029381
Response: 405 Method Not Allowed

**Comment by cpu** [comment]
> I get the same , but my client works with non-staging acme-v02.api.letsencrypt.org

Like @gilbitron mentioned the production API won't have the change that breaks LEClient until Nov 1st 2020.

**Comment by yourivw** [comment]
This issue is now finally resolved in version 1.2.0. This version is now released and implements POST-as-GET instead of the affected GET requests.
